backend/src/main/java/asegroup1/api/models/landregistry/LandRegistryQueryConstraint.java

```diff
@@ -0,0 +1,225 @@
+package asegroup1.api.models.landregistry;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+
+import asegroup1.api.models.landregistry.LandRegistryData.EqualityConstraint;
+
+/**
+ * Contains the constraints to be passed
+ * 
+ * @author Rikkey Paal
+ */
+public class LandRegistryQueryConstraint {
+
+	private HashSet<RangeConstraint> rangeConstraints;
+	private LandRegistryData equalityConstraints;
+	private ArrayList<String> postcodes;
+
+
+	public LandRegistryQueryConstraint(LandRegistryData eqalityConstraints) {
+		rangeConstraints = new HashSet<>();
+		this.equalityConstraints = eqalityConstraints;
+		postcodes = new ArrayList<>();
+	}
+
+	public LandRegistryQueryConstraint() {
+		this(new LandRegistryData());
+	}
+
+
+
+	public LandRegistryData getEqualityConstraints() {
+		return equalityConstraints;
+	}
+
+	/* RANGE CONSTRAINTS */
+
+
+	private void setDateConstraint(boolean isMax, LocalDate date) {
+		rangeConstraints.add(new RangeConstraint("xsd:date", "transactionDate", isMax ? "<" : ">", "\"" + date.toString() + "\""));
+	}
+
+	public void setMaxDate(LocalDate date) {
+		setDateConstraint(true, date);
+	}
+
+	public RangeConstraint getRangeConstraint(String name, String comparitor) {
+		for (RangeConstraint constraint : rangeConstraints) {
+			if (constraint.getName().equals(name) && constraint.getComparitor().equals(comparitor)) {
```

Comment:
`getComparitor` and `setComparitor` should be `getComparator` and `setComparator`. Spelling/typo issue.